contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,272 @@
+    /// @param rateLimitedAddress the address whose buffer will be set
+    /// @param newRateLimitPerSecond the new rate limit per second for this rateLimitedAddress
+    function setIndividualRateLimitPerSecond(
+        address rateLimitedAddress,
+        uint112 newRateLimitPerSecond
+    ) external virtual override onlyGovernor {
+        require(newRateLimitPerSecond <= rateLimitPerSecond, "MultiRateLimited: rateLimitPerSecond too high");
+
+        _updateIndividualBufferStored(rateLimitedAddress);
+        _setIndividualRateLimitPerSecond(rateLimitedAddress, newRateLimitPerSecond);
+    }
+
+    /// @notice set the buffer cap
+    /// @param rateLimitedAddress the address whose buffer will be set
+    /// @param newBufferCap the new buffer cap for this rateLimitedAddress
+    function setIndividualBufferCap(address rateLimitedAddress, uint144 newBufferCap) external virtual override onlyGovernor {
+        require(newBufferCap <= bufferCap, "MultiRateLimited: new buffer cap is over global max");
+
+        _setIndividualBufferCap(rateLimitedAddress, newBufferCap);
+    }
+
+    // ----------- Getters -----------
+
+    /// @notice the amount of action used before hitting limit
+    /// @dev replenishes at rateLimitPerSecond per second up to bufferCap
+    /// @param rateLimitedAddress the address whose buffer will be returned
+    function individualBuffer(address rateLimitedAddress) public view override returns(uint144) {
+        RateLimitData memory rateLimitData = rateLimitPerAddress[rateLimitedAddress];
+
+        uint256 elapsed = block.timestamp - rateLimitData.lastBufferUsedTime;
+        return uint144(Math.min(rateLimitData.bufferStored + (rateLimitData.rateLimitPerSecond * elapsed), rateLimitData.bufferCap));
+    }
+
+    /// @notice the rate per second for each address
+    function getRateLimitPerSecond(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].rateLimitPerSecond;
+    }
+
+    /// @notice the last time the buffer was used by each address
+    function getLastBufferUsedTime(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].lastBufferUsedTime;
+    }
+
+    /// @notice the cap of the buffer that can be used at once
+    function getBufferCap(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].bufferCap;
+    }
+
+    // ----------- Helper Methods -----------
+
+    function _updateAddress(address rateLimitedAddress, uint112 _rateLimitPerSecond, uint144 _bufferCap) internal {
+        uint112 oldRateLimitPerSecond = _rateLimitPerSecond;
+
+        RateLimitData storage rateLimitData = rateLimitPerAddress[rateLimitedAddress];
+
+        require(rateLimitData.lastBufferUsedTime != 0, "MultiRateLimited: rate limit address does not exist");
+        require(_rateLimitPerSecond <= MAX_RATE_LIMIT_PER_SECOND, "MultiRateLimited: rateLimitPerSecond too high");
```

Comment:
The reason we check against global max is because the other maximum is for the non governor, and that is checked in the function that calls this function. `updateAddress` already validates that we are under `maxRateLimitPerSecond` if a non governor is calling.

Reply:
I get it I think, but this is confusing and will be harder to debug/screen for security issues. TBH I'd suggest removing the governor-rate-limit; if the governor can change the minter contract anyways (which it can) then it can bypass the governor-minting-rate-limit. Then we'll only have a non-governor maximum, which is a lot easier to process. In my opinion. @Joeysantoro would love to get your thoughts.